cart/player.go

```diff
@@ -144,6 +148,10 @@ func (player *Player) Update(t int32, world *World, game *Game) {
 		case world.IsItem(tileIndex):
 			world.CollectItem(x, y)
 		}
+		// check if the item was what we wanted
+		if tileIndex == int32(game.DesiredItem) {
```

Comment:
Flagging that, once we add the possibility of desiring multi-block items (if we get to that point), this logic will need updating somehow